src/main/java/seedu/address/logic/commands/ResetCommand.java

```diff
@@ -51,98 +40,24 @@ public ResetCommand(Index index, Tutorial tutorial) {
-        List<String> resetTutorials = new ArrayList<>();
-        String formattedTutorials;
-
-        for (Tutorial tutorial : tutorials) {
-            if (newTutorials.get(tutorial) == AttendanceStatus.NOT_TAKEN_PLACE) {
-                unnecessaryResetTutorials.add(tutorial.tutorial);
-                continue;
-            }
-
-            newTutorials.put(tutorial, AttendanceStatus.NOT_TAKEN_PLACE);
-            resetTutorials.add(tutorial.tutorial);
-        }
-
-        String formattedUnnecessaryTutorials = String.join(", ", unnecessaryResetTutorials);
-
-        if (resetTutorials.isEmpty()) {
-            formattedTutorials = String.format(Messages.MESSAGE_RESET_UNNECESSARY,
-                    personToEdit.getName(), formattedUnnecessaryTutorials);
-        } else {
-            formattedTutorials = generateSuccessMessage(personToEdit, String.join(", ", resetTutorials));
-        }
-
-        return new Pair<Person, String>(
-                new Person(
-                    personToEdit.getName(),
-                    personToEdit.getStudentId(),
-                    personToEdit.getPhone(),
-                    personToEdit.getEmail(),
-                    personToEdit.getTags(),
-                    newTutorials
-                ),
-                formattedTutorials
-        );
+    @Override
+    protected AttendanceStatus getStatus() {
+        return AttendanceStatus.NOT_TAKEN_PLACE;
     }
 
     @Override
-    public CommandResult execute(Model model) throws CommandException {
-        List<Person> currDisplayedList = model.getFilteredPersonList();
-        List<Person> personToEditList = CommandUtil.filterPersonsByIndex(currDisplayedList, index);
-        List<Person> editedPersonList = new ArrayList<>();
-        String finalSuccessMessage = "";
-        for (Person personToEdit : personToEditList) {
-            Pair<Person, String> editedPair = this.generateResetPerson(personToEdit, this.tutorials);
-            Person editedPerson = editedPair.getKey();
-            String formattedTutorials = editedPair.getValue();
-            editedPersonList.add(editedPerson);
-            model.setPerson(personToEdit, editedPerson);
-            String successMessage = formattedTutorials + "\n";
-            finalSuccessMessage += successMessage;
-        }
-        finalSuccessMessage = finalSuccessMessage.trim();
-        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(finalSuccessMessage);
+    protected String getUnnecessaryMessage() {
+        return Messages.MESSAGE_RESET_UNNECESSARY;
     }
 
     /**
      * Generates a command execution success message for the tutorials that were marked as not taken place.
      * {@code personToEdit}.
      */
-    private String generateSuccessMessage(Person personToEdit, String resetTutorials) {
+    protected String generateSuccessMessage(Person personToEdit, String resetTutorials) {
```

Comment:
Should we abstract this out too? Perhaps another method getSuccessMessage() that child classes override and we can put the functionality of generating success messages in the parent class instead.

Reply:
Wow did not see that! Thanks for the suggestion, I have changed refactored the code to include this change. :)